File: Data Structures & Algorithms/majority-element/submission-3.cpp
class Solution {
public:
    int majorityElement(vector<int>& nums) {
        int count=1;
        int val=nums[0];

        for(int i=1;i<nums.size();i++){
            if(val==nums[i]){
                count++;
            }
            else if (count==0){
                count=1;
                val=nums[i];
            }
            else {
                count--;
            }
        }
        return val;
    }
};

File: Data Structures & Algorithms/products-of-array-discluding-self/submission-3.py
class Solution:
    def productExceptSelf(self, nums: List[int]) -> List[int]:
        ans = []
        pro=1
        zero=0
        for num in nums:
            if num==0:
                zero=zero+1
                continue
            
            pro=pro*num
        
        for i in range(len(nums)):
            if zero > 1:
                ans.append(0)
            elif zero==1:
                if nums[i]==0:
                    ans.append(pro)
                else:
                    ans.append(0)
            else:
                ans.append(pro//nums[i])
        
        return ans
